karma.conf.js

```diff
@@ -1,128 +1,43 @@
-            require('karma-firefox-launcher'),
-            require('karma-ie-launcher'),
-            require('karma-sinon-chai'),
-            require('karma-coverage'),
-            require('karma-spec-reporter'),
-            require('karma-sourcemap-loader')
-        ],
-
-        webpackMiddleware: {
-            noInfo: true,
-            stats: false
-        },
-
-        webpack: {
-            devtool: 'inline-source-map',
-
-            resolve: {
-                modules: [
-                    __dirname,
-                    'node_modules'
-                ]
-            },
-
-            module: {
-                rules: [
-                    {
-                        test: /\.js$/,
-                        exclude: /(dist|chai)/,
-                        loader: 'babel-loader',
-                        query: {
-                            presets: ['es2015'],
-                            plugins: [
-                                'transform-flow-strip-types',
-                                'transform-es3-property-literals',
-                                'transform-es3-member-expression-literals',
-                                ['transform-es2015-for-of', {loose: true}],
-                                ['flow-runtime', {
-                                    'assert': true,
-                                    'annotate': true
-                                }],
-                                [ '__coverage__', { only: `${__dirname}/src` } ]
-                            ]
-                        }
-                    },
-                    {
-                        test: /\.(html?|css|json)$/,
-                        loader: 'raw-loader'
-                    }
-                ]
-            },
-            bail: false,
-            plugins: [
-                new webpack.DefinePlugin({
-                    __TEST__: true
-                })
-            ]
+        preprocessors: {
+            'test/index.js': 'webpack'
         },
 
+        webpack: testConfig,
 
+        
         // list of files to exclude
         exclude: [
         ],
 
-
-        // preprocess matching files before serving them to the browser
-        // available preprocessors: https://npmjs.org/browse/keyword/karma-preprocessor
-        preprocessors: {
-            'test/test.js': ['webpack', 'sourcemap'],
-            'test/windows/**/*.js': ['webpack', 'sourcemap'],
-            'src/**/*.js': ['coverage',  'sourcemap']
-        },
-
         // test results reporter to use
         // possible values: 'dots', 'progress'
         // available reporters: https://npmjs.org/browse/keyword/karma-reporter
-        reporters: [
-            'spec',
-            'coverage'
-        ],
-
-        coverageReporter: {
-
-            instrumenterOptions: {
-                istanbul: { noCompact: true }
-            },
-
-            reporters: [
-                {
-                    type: 'text'
-                },
-                {
```

Comment:
Here is the coverage structure: 
![image](https://user-images.githubusercontent.com/3578259/28037737-1049e6d0-6582-11e7-8663-aef1d307d891.png)


Agreed, and I am all about small PR's.  However one thing lead to another here and there really wasn't a good way  to keep it small.